### PR TITLE
Fix IndexOutOfBoundsException in ScriptEngineFactoryHelper

### PR DESCRIPTION
**Pull request by jimtng**
https://community.openhab.org/t/openhab-4-0-snapshot-discussion/142322/319

### CODE REVIEW

**Comment by J-N-K** [comment]
I don't think this is the right way. IMO it's a programming error if we run into that situation and this should not be hidden. I would prefer to throw an `IllegalStateException`.

File: bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineFactoryHelper.java [hunk 1]
```diff
@@ -59,9 +59,13 @@ private ScriptEngineFactoryHelper() {
     }
 
     public static String getPreferredMimeType(ScriptEngineFactory factory) {
-        List<String> mimeTypes = new ArrayList<>(factory.getScriptTypes());
+        List<String> scriptTypes = factory.getScriptTypes();
+        if (scriptTypes.isEmpty()) {
+            throw new IllegalStateException(factory.getClass().getName() + " does not support any scriptTypes.");
```
Comment (J-N-K) [inline bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineFactoryHelper.java:64]:
```suggestion
            throw new IllegalStateException(factory.getClass().getName() + " does not support any scriptTypes. Please report it as a bug.");
```

Comment (jimtng) [reply bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineFactoryHelper.java]: added

**Review by J-N-K — APPROVED** [review]
Thanks

**Pull request merged**